st2common/tests/unit/test_trigger_services.py

```diff
@@ -67,3 +67,63 @@ def test_create_trigger_db_from_rule_use_ref_for_trigger(self):
         trigger_db_ret_2 = trigger_service.create_trigger_db_from_rule(rule_api2)
         self.assertTrue(trigger_db_ret_2 is not None)
         self.assertTrue(trigger_db_ret_2.id == trigger_db_ret_1.id)
+
+    def test_create_or_update_trigger_db_simple_triggers(self):
+        test_fixtures = {
+            'triggertypes': ['triggertype1.json']
+        }
+        loader = FixturesLoader()
+        fixtures = loader.save_fixtures_to_db(fixtures_pack='generic', fixtures_dict=test_fixtures)
+        triggertypes = fixtures['triggertypes']
+        trigger_type_ref = ResourceReference.to_string_reference(
+            name=triggertypes['triggertype1.json']['name'],
+            pack=triggertypes['triggertype1.json']['name'])
+
+        trigger = {
+            'name': 'foo',
+            'pack': 'st2',
+            'type': trigger_type_ref
+        }
+        trigger_service.create_or_update_trigger_db(trigger)
```

Comment:
I assume this function tests create first and then update - right?

it's just hard hard to say, because code references `foo` and `triggertypes['triggertype1.json']['name']` and those values could also be the same for that matter.

If that's indeed the case it might also make sense to add something along the lines of `self.assertNotEqual(triggertypes['triggertype1.json']['name'], 'foo').